### PR TITLE
CPM-996 : add search bar

### PR DESCRIPTION
**Pull request by Beninho**
**Description (for Contributor and Core Developer)**

Add search bar on identifier generators list

Screenshot from 2023-03-13 14-25-00

**Definition Of Done (for Core Developer only)**

- [X] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/identifier-generator/front/src/feature/pages/ListPage.tsx [hunk 1]
```diff
@@ -69,6 +71,14 @@ const ListPage: React.FC<ListPageProps> = ({onCreate}) => {
     return identifierAttributes.find(attribute => attribute.code === target)?.label;
   };
 
+  const filterOnLabelOrCode =
+    (searchValue: string, locale: LocaleCode) =>
+    (entity: {code: string; labels: LabelCollection}): boolean =>
+      -1 !== entity.code.toLowerCase().indexOf(searchValue.toLowerCase()) ||
+      (undefined !== entity.labels[locale] &&
+        -1 !== entity.labels[locale].toLowerCase().indexOf(searchValue.toLowerCase()));
+  const filteredGenerators = generators.filter(filterOnLabelOrCode(search, locale));
```
Comment (cemorin) [inline components/identifier-generator/front/src/feature/pages/ListPage.tsx:80]: Just a suggestion, tell me if it looks clearer or not like this:
```suggestion
  const filteredGenerators = useMemo(() => generators.filter(({code, labels}) => 
    code.toLowerCase().includes(searchValue.toLowerCase()) || labels[locale]?.toLowerCase()?.includes(searchValue.toLowerCase())
  ), [generators, locale, searchValue];
```

File: components/identifier-generator/front/src/feature/pages/__tests__/ListPage.test.tsx [hunk 2]
```diff
@@ -175,4 +175,28 @@ describe('ListPage', () => {
...
+    expect(screen.getByText('Sku generator')).toBeVisible();
+
+    const rows = screen.getAllByRole('row');
+    expect(rows.length).toBe(2);
+
+    fireEvent.change(await screen.findByPlaceholderText('pim_common.search'), {target: {value: 'FOO'}});
+    expect(screen.queryByText('Sku generator')).not.toBeInTheDocument();
+
+    fireEvent.change(await screen.findByPlaceholderText('pim_common.search'), {target: {value: 'TEST'}});
+    expect(screen.getByText('Sku generator')).toBeVisible();
```
Comment (cemorin) [inline components/identifier-generator/front/src/feature/pages/__tests__/ListPage.test.tsx:197]: Are we sure `Sku generator` should be visible here? And maybe add a test to search with code?

Comment (Beninho) [reply components/identifier-generator/front/src/feature/pages/__tests__/ListPage.test.tsx:197]: I will comment my tests. This one is for code. Code is "test"...

Comment (cemorin) [reply components/identifier-generator/front/src/feature/pages/__tests__/ListPage.test.tsx:197]: Indeed, my bad :+1:

File: components/identifier-generator/front/src/feature/pages/__tests__/ListPage.test.tsx [hunk 3]
```diff
@@ -175,4 +175,29 @@ describe('ListPage', () => {
     expect(screen.getByText('Sku generator 0')).toBeVisible();
     expect(screen.getByText('Sku generator 19')).toBeVisible();
   });
+
+  it('should search', async () => {
```
Comment (mmetayer) [inline components/identifier-generator/front/src/feature/pages/__tests__/ListPage.test.tsx:179]:
```suggestion
  it('should search on code or label', async () => {
```

**Review by mmetayer — APPROVED** [review]

**Pull request merged**